### ISSUE TITLE
Get-AzureRmRecoveryServicesAsrProtectableItem fails with "Sequence contains more than one matching element"

### ISSUE THREAD

**Issue by dguertin** [Azure/azure-powershell#7564]
I'm trying to get a list of servers protected/unprotected with ASR, using PowerShell 5.1 and AzureRM 6.10.0. I'm running the commands:
```
$fabric = Get-ASRFabric
$container = Get-ASRProtectionContainer -Fabric $fabric
```
$container is an array of protection containers, and all is fine if I run the command:

`$container[x] | Get-AzureRmRecoveryServicesAsrProtectableItem` 

where x is anything other than the element $container[1]. But $container[1] fails with the error "Sequence contains more than one matching element":

```
$container[1] | Get-AzureRmRecoveryServicesAsrProtectableItem
Get-AzureRmRecoveryServicesAsrProtectableItem : Operation failed.
ClientRequestId: e90029f4-d275-4247-b312-4bfe09ffc20f-2018-10-11 15:16:26Z-Ps
Sequence contains more than one matching element
At line:1 char:17
+ $container[1] | Get-AzureRmRecoveryServicesAsrProtectableItem | ft -P ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmReco...ProtectableItem], Exception
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.GetAzureRmRecoveryServicesAsrProtectableItem
```
What does this mean? I've run the command with the -Debug option, and it produces a lot of output, but nothing that I've been able to help identify where the problem is.

Here's the debug output leading up to the place where it fails:

```
.
.
.
DEBUG: 10/09/2018 16:58:38: 6e843744-d49b-4a9d-86ab-2f6b7adf9edb - AcquireTokenHandlerBase: === Token Acquisition finished successfully. An access token was retuned:
	Access Token Hash: Go4FZX0EvuS+9nmA6aDL4P0l4W2C1Ts7+9nYfDZQEMs=
	Refresh Token Hash: ILb298uPTMiaQP1Nz5fKbM8loO3r2xdQ1KF2qMAZmkU=
	Expiration Time: 10/09/2018 17:25:51 +00:00
	User Hash: logWkqU98C9nUpUfZGojURITpKuVEuGhDdIraGjYouk=
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2018 16:58:38:  - TokenCache: Serializing token cache with 3 items.
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/Subscriptions/b1d4fdad-b17a-42e0-a6db-98bac63fxxx/resourceGroups/mygroup/providers/Microsoft.RecoveryServices/vaults/myvault/replicationFabrics/0642e13e08e6c8ee269d34c99004a28eb86d73889096242b132bc1f76bf9xxxx/replicationProtectionContainers/952f4ef5-5638-497b-ae97-a94c5be4xxxx/replicationProtectableItems?api-version=2018-01-10&%24skipToken=ReplicationGroup:Begin

Headers:
x-ms-client-request-id        : 7105b902-b1ab-4de0-9fd7-283410bfddf7-2018-10-09 16:58:38Z-Ps
accept-language               : en-US
Agent-Authentication          : {"NotBeforeTimestamp":"\/Date(1539100718661)\/","NotAfterTimestamp":"\/Date(1539705518661)\/","ClientRequestId":"7105b902-b1ab-4de0-9fd7-283410bfddf7-2018-10-09 16:58:38Z-Ps","HashFunction":"HMACSHA256","Hmac":"RKpXH3UQoH9AgIhg1+JKXDxnAWinI2McsrjMyJ9YHLQ=","Version":{"Major":1,"Minor":2,"Build":-1,"Revision":-1,"MajorRevision":-1,"MinorRevision":-1},"PropertyBag":{}}

Body:
DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Pragma                        : no-cache
Cache-Control                 : no-cache
Server                        : Microsoft-IIS/10.0,Microsoft-IIS/10.0
x-ms-request-id               : 740f251e-1be3-4e90-b746-56bcd6f4a3f2 10/9/2018 4:58:39 PM
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET
Strict-Transport-Security     : max-age=31536000; includeSubDomains
X-Content-Type-Options        : nosniff
x-ms-client-request-id        : 740f251e-1be3-4e90-b746-56bcd6f4a3f2
x-ms-ratelimit-remaining-subscription-reads: 11999
x-ms-correlation-request-id   : 16520d28-3709-47c4-91a7-3424056af1db
x-ms-routing-request-id       : WESTEUROPE:20181009T165839Z:16520d28-3709-47c4-91a7-3424056af1db
Date                          : Tue, 09 Oct 2018 16:58:38 GMT

Body:
{
  "value": [],
  "nextLink": null
}
Confirm
Operation failed.
ClientRequestId: 7105b902-b1ab-4de0-9fd7-283410bfddf7-2018-10-09 16:58:38Z-Ps
Sequence contains more than one matching element
&Yes  Yes to &All  &Halt Command  &Suspend
A
Get-AzureRmRecoveryServicesAsrProtectableItem : Operation failed.
ClientRequestId: 7105b902-b1ab-4de0-9fd7-283410bfddf7-2018-10-09 16:58:38Z-Ps
Sequence contains more than one matching element
At line:1 char:17
+ $container[1] | Get-AzureRmRecoveryServicesAsrProtectableItem -Debug
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmReco...ProtectableItem], Exception
    + FullyQualifiedErrorId : 
Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.GetAzureRmRecoveryServicesAsrProtectableItem
Get-AzureRmRecoveryServicesAsrProtectableItem : Operation failed.
ClientRequestId: 7105b902-b1ab-4de0-9fd7-283410bfddf7-2018-10-09 16:58:38Z-Ps
Sequence contains more than one matching element
At line:1 char:17
+ $container[1] | Get-AzureRmRecoveryServicesAsrProtectableItem -Debug
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmReco...ProtectableItem], Exception
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.GetAzureRmRecoveryServicesAsrProtectableItem

DEBUG: AzureQoSEvent: CommandName - Get-AzureRmRecoveryServicesAsrProtectableItem; IsSuccess - False; Duration - 00:00:07.7724629; Exception - System.Exception: Operation failed.
ClientRequestId: 7105b902-b1ab-4de0-9fd7-283410bfddf7-2018-10-09 16:58:38Z-Ps
Sequence contains more than one matching element ---> System.InvalidOperationException: Sequence contains more than one matching element
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.ASRProtectableItem.UpdateDiskDetails(IList`1 diskDetails)
   at Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.ASRProtectableItem..ctor(ProtectableItem pi)
   at Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.GetAzureRmRecoveryServicesAsrProtectableItem.<>c.<WriteProtectableItems>b__17_0(ProtectableItem pi)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Management.Automation.MshCommandRuntime._WriteObjectsSkipAllowCheck(Object sendToPipeline)
   at System.Security.SecurityContext.Run(SecurityContext securityContext, ContextCallback callback, Object state)
   at System.Management.Automation.MshCommandRuntime.WriteObject(Object sendToPipeline, Boolean enumerateCollection)
   at System.Management.Automation.Cmdlet.WriteObject(Object sendToPipeline, Boolean enumerateCollection)
   at Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.GetAzureRmRecoveryServicesAsrProtectableItem.WriteProtectableItems(IList`1 protectableItems)
   at Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.SiteRecoveryCmdletBase.ExecuteCmdlet()
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.SiteRecoveryCmdletBase.HandleException(Exception ex)
   at Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.SiteRecoveryCmdletBase.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 12:58:42 PM - GetAzureRmRecoveryServicesAsrProtectableItem end processing.
DEBUG: 12:58:42 PM - GetAzureRmRecoveryServicesAsrProtectableItem end processing.
PS C:\WINDOWS\system32> Stop-Transcript
**********************
Windows PowerShell transcript end
End time: 20181009125845
**********************
```
 How can I try to locate the source of the problem in $container[1]?

(I didn't file this as a bug report because I don't know if it's a bug, but if someone decides it is, I'll resubmit it in the desired format.)

**Comment by MiYanni** [comment]
@viverm Can you take a look at this? It looks like your constructor for `ASRProtectableItem` is expecting there to only be one of something, but in @dguertin's case, it is returning more than one; so, the call to `SingleOrDefault` fails. Did you mean to use `FirstOrDefault` instead?

**Comment by dguertin** [comment]
@viverm You nailed it. That was exactly the problem. We had one VM that had been erroneously configured with two disks both set as OS disks. All I needed to do was uncheck the "contains the operating system for the virtual machine" box for one of the disks, and the problem was solved.

We had 155 VMs in that container, and it was pretty tedious to find the offending VM, but maybe there's an easy way that I missed.

Perhaps an error message about "more than one OS disk" would make this easier to debug.

Thanks for your help.
